consensus/spos/bls/blsWorker_test.go

```diff
@@ -23,8 +25,66 @@ func initConsensusState() *spos.ConsensusState {
 	return initConsensusStateWithKeysHandler(&testscommon.KeysHandlerStub{})
 }
 
+func initConsensusStateWithArgs(keysHandler consensus.KeysHandler, consensusGroupSize int, mapKeys map[string]crypto.PrivateKey) *spos.ConsensusState {
+	return initConsensusStateWithKeysHandlerWithGroupSizeWithRealKeys(keysHandler, consensusGroupSize, mapKeys)
+}
+
 func initConsensusStateWithKeysHandler(keysHandler consensus.KeysHandler) *spos.ConsensusState {
 	consensusGroupSize := 9
+	return initConsensusStateWithKeysHandlerWithGroupSize(keysHandler, consensusGroupSize)
+}
+
+func createEligibleListFromMap(mapKeys map[string]crypto.PrivateKey) []string {
+	eligibleList := make([]string, 0, len(mapKeys))
+	for key := range mapKeys {
+		eligibleList = append(eligibleList, key)
+	}
+	return eligibleList
```

Comment:
maybe add a sorting on the eligibleList, so that we always have the same ordering if called multiple times for the same keys.